### PR TITLE
Add ecosystem test for homeassistant

### PR DESCRIPTION
**Pull request by zanieb**
Extends #2942 with another repository.

This one might be too big it takes 5 minutes haha — but it's faster than Windows still!

### CODE REVIEW

**Review by charliermarsh — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -226,22 +226,36 @@ jobs:
...
+          - {
+              repo: "prefecthq/prefect",
+              command: "uv pip install --upgrade -e '.[dev]'",
+              python: "3.9",
+            }
+          - {
+              repo: "home-assistant/core",
+              command: "uv pip install -r requirements_all.txt",
+              python: "3.12",
+            }
```
Comment (konstin) [inline .github/workflows/ci.yml:243]: I think we can skip the parentheses

```suggestion
          - repo: "prefecthq/prefect",
            command: "uv pip install --upgrade -e '.[dev]'",
            python: "3.9",
          - repo: "home-assistant/core",
            command: "uv pip install -r requirements_all.txt",
            python: "3.12",
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -226,22 +226,36 @@ jobs:
...
+      fail-fast: false
     steps:
       - uses: actions/checkout@v4
```
Comment (konstin) [inline .github/workflows/ci.yml:246]: Do we need to perform a checkout?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -226,22 +226,36 @@ jobs:
...
+          python-version: ${{ matrix.python }}
 
       - name: "Download binary"
         uses: actions/download-artifact@v4
         with:
           name: uv-linux-${{ github.sha }}
 
       - name: "Clone project"
-        run: gh repo clone prefecthq/prefect
+        run: gh repo clone ${{ matrix.repo }} repo
```
Comment (konstin) [inline .github/workflows/ci.yml:258]: We can speed this up with depth 1, `actions/checkout@v4` should do this by default

File: .github/workflows/ci.yml (hunk 2 above)
Comment (zanieb) [reply .github/workflows/ci.yml:246]: Nope! :D

```suggestion
```

**Comment by konstin** [comment]
Home assistant uses a custom index, no build and a musl based python 3.12 for their deploys, i think we need to replicate this if we want to use it as a test

**Pull request closed without merging**